In [1]:
import random
import math

size = 100000
punch = 0
for i in range(size):
    x1 = random.uniform(0,1)
    y1 = random.uniform(0,1)
    x2 = random.uniform(0,1)
    y2 = random.uniform(0,1)
    # center
    x = (x1+x2)/2
    y = (y1+y2)/2
    # radius
    r = math.sqrt((x-x1)**2 + (y-y1)**2)
    if (x+r > 1) or (x-r < 0) or (y+r > 1) or (y-r < 0):
        punch += 1

result = punch/size
print(result)

0.47661


Trivially, the result is relevant with pi

Since what we lookfing for is the probability outside of the square, it should be a complement of (k/pi) or (pi/k)

In [2]:
target = 1-result
print(target/ math.pi)
print(target * math.pi)

0.1666002113297342
1.644278178962362


0.1666 obviously is 1/6

## As a result, the answer we looking for is 1=pi/6